### PR TITLE
HoC 2024 - Update OpenGraph for November launch

### PR DESCRIPTION
**Pull request by kelbyhawn**
Updates the following OpenGraph tags:
- Title and description on https://hourofcode.com
  - Also applies OpenGraph site-wide!
- New Music Lab: Jam Session image and description on https://code.org/music
- Default generic description on https://code.org/minecraft

I also cleaned up `social_metadata.rb` by removing outdated entries and removed hoc_mode logic since it's not used. 

## Links
Jira ticket: [ACQ-2542](https://codedotorg.atlassian.net/browse/ACQ-2542) 

## Testing story
Tested locally

### CODE REVIEW

File: pegasus/src/social_metadata.rb [hunk 1]
```diff
@@ -86,47 +50,15 @@ def get_social_metadata_for_page(request)
     },
     "hourofcode.com" => {
       "default" => {
-        title: hoc_s("hoc_2024.social.title"),
-        description: hoc_s("hoc_2024.social.desc"),
+        title: hoc_s("hoc_2024.social.v2.title"),
+        description: hoc_s("hoc_2024.social.v2.desc"),
```
Comment (kelbyhawn) [inline pegasus/src/social_metadata.rb:54]: hoc

Comment (kelbyhawn) [inline pegasus/src/social_metadata.rb:54]: hoc

File: pegasus/src/social_metadata.rb [hunk 2]
```diff
@@ -86,47 +50,15 @@ def get_social_metadata_for_page(request)
...
-      },
-      "soon-hoc-dance" => {
-        title: hoc_s(:tutorial_mchoc_name),
-        description: hoc_s(:social_hoc2018_mc_creativity),
-        image: images[:mc_social_2018]
-      },
       "default" => {
         title: hoc_s(:tutorial_mchoc_name),
-        description: hoc_s(:social_hoc2018_mc_creativity),
+        description: hoc_s("hoc_page.activities.minecraft.desc"),
```
Comment (kelbyhawn) [inline pegasus/src/social_metadata.rb:61]: minecraft

File: pegasus/src/social_metadata.rb [hunk 3]
```diff
@@ -210,8 +128,8 @@ def get_social_metadata_for_page(request)
     "music_lab" => {
       "default" => {
         title: hoc_s("music_lab.opengraph_title", markdown: :inline, locals: {music_lab: "Music Lab"}),
-        description: hoc_s("music_lab.banner.desc"),
-        image: images[:music_lab]
+        description: hoc_s("music_lab.jam_session.banner.desc_01"),
+        image: images[:music_lab_jam]
```
Comment (kelbyhawn) [inline pegasus/src/social_metadata.rb:132]: music

**Review by drizco — APPROVED** [review]
lgtm!

**Pull request merged**